packages/gatsby/src/redux/reducers/pages.ts

```diff
@@ -1,12 +1,21 @@
-const normalize = require(`normalize-path`)
+import normalize from "normalize-path"
+import {
+  IGatsbyState,
+  IDeleteCacheAction,
+  ICreatePageAction,
+  IDeletePageAction,
+} from "../types"
 
-module.exports = (state = new Map(), action) => {
+export const pagesReducer = (
+  state: IGatsbyState["pages"] = new Map(),
+  action: IDeleteCacheAction | ICreatePageAction | IDeletePageAction
+): IGatsbyState["pages"] => {
   switch (action.type) {
     case `DELETE_CACHE`:
       return new Map()
     case `CREATE_PAGE`: {
       action.payload.component = normalize(action.payload.component)
-      if (!action.plugin && !action.plugin.name) {
+      if (!action.plugin || !action.plugin.name) {
```

Comment:
This is changing behavior, why was that changed?

Reply:
With the original code, the `!action.plugin.name` was short-circuited and TypeScript was saying that `action.plugin.name not assignable to parameter of type never`.

Reply:
Ah, yeah, sure! We can use optional chaining here, so `!action.plugin?.name`

Reply:
So it would look something like `if (!action.plugin && !action.plugin?.name)` ?

Reply:
It would be `if(!action.plugin?.name)` - you can read more here: https://github.com/tc39/proposal-optional-chaining#overview-and-motivation

Reply:
Whoa! This is really cool. Thanks for the information, the commit should be up soon.